apps/ftu/js/late_customization.js

```diff
@@ -46,20 +56,17 @@
       window.addEventListener('online', this);
       window.addEventListener('offline', this);
 
+      this._operatorInfo = this.getInfoFromMobileConnection();
       this.waitForSettingValues().then((results) => {
-        var [url, info] = results;
+        var [url, predefinedInfo] = results;
         if (url) {
           this._customizationManifestUrl = url;
         }
-        if (info && info.operator) {
-          this._operatorInfo = info;
-          if (info.operator && !info.carrier) {
-            info.carrier = info.operator;
-          }
+        if (predefinedInfo) {
+          // store operator/params separate from
+          // bona-fide mobileconnection details
+          this._predefinedOperatorInfo = predefinedInfo;
         }
-        // skip waiting for operator info from mobileConnections if we got
-        // it from settings
-        return this._operatorInfo || this.waitForOperatorInfo();
       }).then((results) => {
```

Comment:
This results variable now should be undefined, we are not using it, before we were just waiting for the promise to finish.

Which means the previous `then` block is already synchronous, so we could potentially group the two `then` blocks into a single one, not having to wait for the next tick.